src/store/diagnostics.ts

```diff
@@ -97,12 +122,29 @@ export const actions = {
 
     rowToUpdate.mute = isMuted;
 
+    await dispatch(
+      'preferences/updatePreferencesData',
+      {
+        property: 'diagnostics.mutedChecks',
+        value:    {
+          ...rootState.preferences.preferences.diagnostics.mutedChecks,
+          [rowToUpdate.id]: isMuted,
+        } as Record<string, boolean>,
```

Comment:
Any reason we can't just do
```
property: `diagnostics.mutedChecks.${rowToUpdate.id}`
```

Reply:
I mean, you can if want to express the same thing in a more succinct way 😆